### ISSUE TITLE
ALP preloading appears to be broken

### ISSUE THREAD

**Issue by lexandera** [ampproject/amphtml#9262]
While trying to implement ALP support in ad clickthroughs I was unable to get the behavior described in [this document](https://github.com/jasti/amp-ads-testing/blob/master/dfp-amp-testing/amp_tests/playground/landingpages/ALP%20Docs/alp.md).

Instead of seeing a document request on touch start, the following warning is printed in the browser console: `<link rel=preload> must have a valid "as" value`. On touch release the browser window navigates to the linked landing page.

The same behavior can be observed in [this example](https://dfp-amp-testing-1185.appspot.com/amp_tests/amp-alp-demo.html) (also linked from the document above) and in `/examples/alp/creative.html`.

I also tried searching for the example document on Google search; and when [that page is loaded in the search page AMP viewer](https://www.google.com/amp/dfp-amp-testing-1185.appspot.com/amp_tests/amp-alp-demo.html), the clickthrough to the landing page doesn't work at all - there is no redirect away from the demo page.

**Comment by zhouyx** [comment]
I found a related discussion in #2492. Chrome seems to the take `as=document` as an invalid value, though it should be a valid value according to [spec](https://w3c.github.io/preload/#link-element-interface-extensions)

I found one example from the spec. Chrome throws `<link rel=preload> must have a valid "as" value` warning, while Safari doesn't.
```
EXAMPLE 8
<script>
var res = document.createElement("link");
res.rel = "preload";
res.as = "document";
res.href = "/other/widget.html";
document.head.appendChild(res);
</script>
```

**Comment by lexandera** [comment]
Is ALP supposed to be supported in A4A? And does it still require alp.js to be included, or is it supposed to just work automatically? I couldn't find any documentation related to this, and validation fails for A4A documents containing a script tag that loads alp.js.

**Comment by cramforce** [comment]
There is some churn on the spec lately. `document` as a preload value is currently not supported. The right way today is to use both no value and the value `fetch`.

**Comment by lannka** [comment]
this should have been addressed in #9823 
though i guess the browser side support has just been [checked in a month ago](https://github.com/w3c/preload/issues/80#issuecomment-305915330), so right we still see the [same error message on chrome stable](http://output.jsbin.com/newanaf
) (couldn't launch chrome canary on my machine to verify, it crashes on every start up!).
